### DISCUSSION TITLE
Liquid Exception: Failed to parse BibTeX on value "pdf""

### DISCUSSION THREAD

**Discussion by uverma** [alshedivat/al-folio#242] [Q&A] [answered]
I am unable to deploy the website. The Github Action fails with the error "Liquid Exception: Failed to parse BibTeX on value "pdf"" 

The source code is available on https://github.com/uverma/uverma.github.io 

Run yes | bin/deploy --verbose  --src source --deploy master
+ shift
Deploying...
+ [[ 4 > 0 ]]
+ key=--src
+ case $key in
+ SRC_BRANCH=source
+ shift
+ shift
+ [[ 2 > 0 ]]
+ key=--deploy
+ case $key in
+ DEPLOY_BRANCH=master
+ shift
+ shift
+ [[ 0 > 0 ]]
+ set -e
+ echo Deploying...
+ echo 'Source branch: source'
+ echo 'Deploy branch: master'
Source branch: source
Deploy branch: master
+ read -r -p 'Do you want to proceed? [y/N] ' response
+ [[ ! y =~ ^([yY][eE][sS]|[yY])+$ ]]
+ git diff-index --quiet HEAD --
++ git ls-files --exclude-standard --others
+ test -z ''
++ git branch
++ grep source
++ tr ' ' '\n'
++ tail -1
+ '[' source ']'
+ git checkout source
Already on 'source'
Your branch is up to date with 'origin/source'.
++ git branch
++ grep master
+ '[' ']'
+ git checkout -b master
Switched to a new branch 'master'
+ bundle exec jekyll build
Configuration file: /home/runner/work/uverma.github.io/uverma.github.io/_config.yml
            Source: /home/runner/work/uverma.github.io/uverma.github.io
       Destination: /home/runner/work/uverma.github.io/uverma.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
         AutoPages: Disabled/Not configured in site.config.
        Pagination: Complete, processed 1 pagination page(s)
E, [2021-04-11 08:36:51#2513] ERROR -- : Failed to parse BibTeX on value "pdf" (NAME) [#<BibTeX::Bibliography data=[1]>, "@", #<BibTeX::Entry >, {:title=>["Weed Density and Distribution Estimation for Precision Agriculture Using Semi-Supervised Learning"], :author=>["Shorewala, Shantam and Ashfaque, Armaan and Sidharth, R and Verma, Ujjwal"], :journal=>["IEEE Access"], :volume=>["9"], :pages=>["27971--27986"], :year=>["2021"], :publisher=>["IEEE"], :abstract=>["Uncontrolled growth of weeds can severely affect the crop yield and quality. Unrestricted use of herbicide for weed removal alters biodiversity and cause environmental pollution. Instead, identifying weed-infested regions can aid selective chemical treatment of these regions. Advances in analyzing farm images have resulted in solutions to identify weed plants. However, a majority of these approaches are based on supervised learning methods which requires huge amount of manually annotated images. As a result, these supervised approaches are economically infeasible for the individual farmer because of the wide variety of plant species being cultivated. In this paper, we propose a deep learning-based semi-supervised approach for robust estimation of weed density and distribution across farmlands using only limited color images acquired from autonomous robots. This weed density and distribution can be useful in a site-specific weed management system for selective treatment of infected areas using autonomous robots. In this work, the foreground vegetation pixels containing crops and weeds are first identified using a Convolutional Neural Network (CNN) based unsupervised segmentation. Subsequently, the weed infected regions are identified using a fine-tuned CNN, eliminating the need for designing hand-crafted features. The approach is validated on two datasets of different crop/weed species (1) Crop Weed Field Image Dataset (CWFID), which consists of carrot plant images and the (2) Sugar Beets dataset. The proposed method is able to localize weed-infested regions a maximum recall of 0.99 and estimate weed density with a maximum accuracy of 82.13%. Hence, the proposed approach is shown to generalize to different plant species without the need for extensive labeled data."]}]
  Liquid Exception: Failed to parse BibTeX on value "pdf" (NAME) [#<BibTeX::Bibliography data=[1]>, "@", #<BibTeX::Entry >, {:title=>["Weed Density and Distribution Estimation for Precision Agriculture Using Semi-Supervised Learning"], :author=>["Shorewala, Shantam and Ashfaque, Armaan and Sidharth, R and Verma, Ujjwal"], :journal=>["IEEE Access"], :volume=>["9"], :pages=>["27971--27986"], :year=>["2021"], :publisher=>["IEEE"], :abstract=>["Uncontrolled growth of weeds can severely affect the crop yield and quality. Unrestricted use of herbicide for weed removal alters biodiversity and cause environmental pollution. Instead, identifying weed-infested regions can aid selective chemical treatment of these regions. Advances in analyzing farm images have resulted in solutions to identify weed plants. However, a majority of these approaches are based on supervised learning methods which requires huge amount of manually annotated images. As a result, these supervised approaches are economically infeasible for the individual farmer because of the wide variety of plant species being cultivated. In this paper, we propose a deep learning-based semi-supervised approach for robust estimation of weed density and distribution across farmlands using only limited color images acquired from autonomous robots. This weed density and distribution can be useful in a site-specific weed management system for selective treatment of infected areas using autonomous robots. In this work, the foreground vegetation pixels containing crops and weeds are first identified using a Convolutional Neural Network (CNN) based unsupervised segmentation. Subsequently, the weed infected regions are identified using a fine-tuned CNN, eliminating the need for designing hand-crafted features. The approach is validated on two datasets of different crop/weed species (1) Crop Weed Field Image Dataset (CWFID), which consists of carrot plant images and the (2) Sugar Beets dataset. The proposed method is able to localize weed-infested regions a maximum recall of 0.99 and estimate weed density with a maximum accuracy of 82.13%. Hence, the proposed approach is shown to generalize to different plant species without the need for extensive labeled data."]}] in /home/runner/work/uverma.github.io/uverma.github.io/_layouts/about.html
                    ------------------------------------------------
      Jekyll 4.2.0   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
bibtex.y:138:in `on_error': Failed to parse BibTeX on value "pdf" (NAME) [#<BibTeX::Bibliography data=[1]>, "@", #<BibTeX::Entry >, {:title=>["Weed Density and Distribution Estimation for Precision Agriculture Using Semi-Supervised Learning"], :author=>["Shorewala, Shantam and Ashfaque, Armaan and Sidharth, R and Verma, Ujjwal"], :journal=>["IEEE Access"], :volume=>["9"], :pages=>["27971--27986"], :year=>["2021"], :publisher=>["IEEE"], :abstract=>["Uncontrolled growth of weeds can severely affect the crop yield and quality. Unrestricted use of herbicide for weed removal alters biodiversity and cause environmental pollution. Instead, identifying weed-infested regions can aid selective chemical treatment of these regions. Advances in analyzing farm images have resulted in solutions to identify weed plants. However, a majority of these approaches are based on supervised learning methods which requires huge amount of manually annotated images. As a result, these supervised approaches are economically infeasible for the individual farmer because of the wide variety of plant species being cultivated. In this paper, we propose a deep learning-based semi-supervised approach for robust estimation of weed density and distribution across farmlands using only limited color images acquired from autonomous robots. This weed density and distribution can be useful in a site-specific weed management system for selective treatment of infected areas using autonomous robots. In this work, the foreground vegetation pixels containing crops and weeds are first identified using a Convolutional Neural Network (CNN) based unsupervised segmentation. Subsequently, the weed infected regions are identified using a fine-tuned CNN, eliminating the need for designing hand-crafted features. The approach is validated on two datasets of different crop/weed species (1) Crop Weed Field Image Dataset (CWFID), which consists of carrot plant images and the (2) Sugar Beets dataset. The proposed method is able to localize weed-infested regions a maximum recall of 0.99 and estimate weed density with a maximum accuracy of 82.13%. Hence, the proposed approach is shown to generalize to different plant species without the need for extensive labeled data."]}] (BibTeX::ParseError)

**Comment by alshedivat** [comment, marked as answer]
Looks like one of your bibtex entries is incorrectly formatted. I believe you need to add a comma at the end of [line 42](https://github.com/uverma/uverma.github.io/blob/source/_bibliography/papers.bib#L42). Hope this helps.
